### PR TITLE
Revert "Revert "Add NetSuite::ProfilesSorter""

### PR DESCRIPTION
**Pull request by bobbytables**
Reverts namely/connect#268

REVERT-CEPTION. :fire: :fire: :fire: :fire:

### CODE REVIEW

File: app/services/net_suite/profiles_sorter.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    end
+
+    def call
+      profiles.reduce([]) do |sorted_profiles, profile|
+        report_to = ReportTo.for(profile)
+        sorted_profiles.take_while do |p|
+          p.id != report_to.id
+        end + [profile] + sorted_profiles.drop_while do |p|
+          p.id != report_to.id
+        end
```
Comment (bobbytables) [inline app/services/net_suite/profiles_sorter.rb]: Here's where we'll need to append the `supervisor_netsuite_id` to the profile.

File: app/services/net_suite/profiles_sorter.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    class ProfileWithSupervisorId < SimpleDelegator
+      def initialize(profile:, reports_to:)
+        @reports_to = reports_to
+        super(profile)
+      end
+
+      def class
+        __getobj__.class
+      end
```
Comment (bobbytables) [inline app/services/net_suite/profiles_sorter.rb]: Interesting. Why is this here?

Comment (cored) [inline app/services/net_suite/profiles_sorter.rb]: @bobbytables It just osd on my part. So according to the gang of four the decorator should have be the same class as the underlying object that you want to expand the public API upon; that's why that is there. But right now is not needed.

Comment (bobbytables) [inline app/services/net_suite/profiles_sorter.rb]: I think if we're going to introduce code like this we should at a minimum document the method with a comment explaining it's purpose.

But I'm also not the biggest fan of not introducing code that isn't used :/

Comment (cored) [inline app/services/net_suite/profiles_sorter.rb]: You know what, I will remove this. It doesn't give any value.
